src/commands/moderation/clearMessages.ts

```diff
@@ -0,0 +1,109 @@
+    }
+
+    const confirmEmbed = makeEmbed({
+        title: 'Confirm or Cancel',
+        description: `You are about to clear **${amount}** message${amount > 1 ? 's' : ''}. This action **cannot** be undone.`,
+        color: Colors.Red,
+        footer: { text: 'Please confirm or cancel this action within 2 minutes.' },
+    });
+
+    const response = await interaction.reply({
+        embeds: [confirmEmbed],
+        components: [buttonRow],
+        ephemeral: true,
+    });
+
+    const filter = (buttonInteraction: Interaction) => buttonInteraction.user.id === interaction.user.id;
+
+    try {
+        const confirmation = await response.awaitMessageComponent({ filter, time: 120_000 });
+        if (confirmation.customId === 'clearMessages_confirm') {
+            try {
+                const messages = await (channel as TextChannel).bulkDelete(amount, true);
+                const replyEmbed = makeEmbed({
+                    title: 'Messages Cleared',
+                    description: `Successfully cleared **${messages.size}** message${amount > 1 ? 's' : ''}.`,
+                    color: Colors.Green,
+                    timestamp: new Date(),
+                });
+
+                const modLogsChannel = interaction.guild.channels.resolve(constantsConfig.channels.MOD_LOGS) as TextChannel;
+                const modLogEmbed = makeEmbed({
+                    title: '🧹 Messages Cleared',
+                    description: 'Messages have been cleared.',
+                    color: Colors.Green,
+                    fields: [
+                        { name: 'Moderator', value: `<@${user.id}>`, inline: true },
+                        { name: 'Channel', value: `<#${channel.id}>`, inline: true },
+                        { name: 'Amount', value: `${messages.size}`, inline: true },
+                    ],
+                    footer: { text: `Moderator ID: ${user.id}`, iconURL: user.displayAvatarURL() },
+                    timestamp: new Date(),
+                });
+
+                try {
+                    await modLogsChannel.send({ embeds: [modLogEmbed] });
+                } catch (e) {
+                    Logger.error('An error occurred while trying to send the mod log:', e);
+                }
+                setTimeout(async () => {
+                    try {
+                        return interaction.deleteReply();
+                    } catch (error) {
+                        Logger.error('Failed to delete the reply message:', error);
+                        return interaction.editReply({ content: 'Failed to delete the reply message.' });
+                    }
+                }, 5000);
+                return interaction.editReply({ content: '', embeds: [replyEmbed], components: [] });
+            } catch (error) {
+                Logger.error('Error clearing messages:', error);
+                return interaction.editReply({ content: 'There was an error trying to clear messages in this channel.' });
```

Comment:
Mini improvement for better UX.
```suggestion
                return interaction.editReply({ content: 'There was an error trying to clear messages in this channel. The error has been logged.' });
```